### PR TITLE
Price timeout

### PR DESCRIPTION
**Pull request by anxolin**
Part of #780 

This PR adds a timeout for APIs that takes too long to complete. If it takes more than 10 seconds, it for any of the APIs, it will try to get by with the results of the other price estimations.

This PR also improves the logged information. I highlighted in the screenshot the new messages. Basically now you can see:
- The details of the failed price queries. Before it was failing silently! Very bad... 
- It also shows the name of the source that failed
- It shows the name of the sources that succeed
- It show the winning price (or prices if they have the same)

image

### CODE REVIEW

**Comment by biocom** [comment]
Seems to work. You think it's low-effort to trigger `useIsQuoteRefreshing()` so that while it is fetching prices from the API's, it shows the COW loader? Right now there's no visual feedback.

File: src/custom/hooks/useRefetchPriceCallback.ts
Comment (biocom) [inline src/custom/hooks/useRefetchPriceCallback.ts:100]: You could consider listing all our price end-points with `console.table()` to provide a nice visual overview of all end-points, prices and the last column 'WINNING PRICE' with a ✅.

**Review by biocom — APPROVED** [review]
Approve with some comments.

**Comment by anxolin** [comment]
David, u can approve after merge. we want to cut a version with this :)

File: src/custom/hooks/useRefetchPriceCallback.ts
Comment (anxolin) [reply src/custom/hooks/useRefetchPriceCallback.ts:100]: good idea, will do in new PR

File: src/custom/hooks/useRefetchPriceCallback.ts
Comment (alfetopito) [inline src/custom/hooks/useRefetchPriceCallback.ts:114]: Didn't get why the filter here

Comment (anxolin) [reply src/custom/hooks/useRefetchPriceCallback.ts:114]: So we know the winning amount, now we want to know who is the winning price (or winning prices, is there's more than one with the same price)

Before u say so, im not concern with trying to get the amount and winning prices in one round by using a reduce function. I like the simplicity on line 109 (using the lib native max/min function)

File: src/custom/hooks/useRefetchPriceCallback.ts
Comment (anxolin) [reply src/custom/hooks/useRefetchPriceCallback.ts:100]: https://github.com/gnosis/cowswap/issues/841

File: src/custom/hooks/useRefetchPriceCallback.ts
Comment (alfetopito) [reply src/custom/hooks/useRefetchPriceCallback.ts:114]: lol, not saying anything =P

Comment (anxolin) [reply src/custom/hooks/useRefetchPriceCallback.ts:114]: 😂

**Pull request merged**
